File: example/src/test/java/ru/virgil/example/util/security/user/WithMockFirebaseUser.java
package ru.virgil.example.util.security.user;

import org.springframework.security.test.context.support.TestExecutionEvent;
import org.springframework.security.test.context.support.WithSecurityContext;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Retention(RetentionPolicy.RUNTIME)
@WithSecurityContext(
        factory = MockUserSecurityContextFactory.class,
        setupBefore = TestExecutionEvent.TEST_METHOD
)
public @interface WithMockFirebaseUser {

    String firebaseUserId() default "";

    String firebaseAuthToken() default "";

}

File: example/src/test/java/ru/virgil/example/util/security/user/WithMockFirebaseUser.kt
package ru.virgil.example.util.security.user

import org.springframework.security.test.context.support.TestExecutionEvent
import org.springframework.security.test.context.support.WithSecurityContext
import ru.virgil.example.security.UserAuthority

@Retention(AnnotationRetention.RUNTIME)
@WithSecurityContext(factory = MockUserSecurityContextFactory::class, setupBefore = TestExecutionEvent.TEST_METHOD)
annotation class WithMockFirebaseUser(
    val authorities: Array<UserAuthority> = [UserAuthority.ROLE_USER],
    val firebaseUserId: String = "user-id",
    val firebaseAuthToken: String = "user-auth-token",
)
